store/postgres/src/relational_queries.rs

```diff
@@ -953,9 +985,73 @@ impl<'a> QueryFilter<'a> {
         QueryFilter {
             filter,
             table: self.table,
+            layout: self.layout,
+            table_prefix: self.table_prefix.clone(),
         }
     }
 
+    fn child(
+        &self,
+        attribute: &Attribute,
+        entity_type: &'a EntityType,
+        filter: &'a EntityFilter,
+        mut out: AstPass<Pg>,
+    ) -> QueryResult<()> {
+        let inner_table = self
+            .layout
+            .table_for_entity(entity_type)
+            .expect("Table for child entity not found");
+
+        let inner_prefix = "inner.";
+
+        out.push_sql("exists (select 1");
+
+        out.push_sql(" from ");
+        out.push_sql(inner_table.qualified_name.as_str());
+        out.push_sql(" inner");
+
+        out.push_sql(" where ");
+
+        // Join parent and inner table
+        let parent_column = self
+            .table
+            .column_for_field(attribute)
+            .expect("Column for an attribute not found");
+        let parent_prefix = "c.";
+        let inner_column = inner_table.primary_key();
+
+        out.push_sql(inner_prefix);
+        out.push_sql(inner_column.name.as_str());
+        out.push_sql(" = ");
+        out.push_sql(parent_prefix);
+        out.push_sql(parent_column.name.as_str());
+
+        out.push_sql(" and ");
+
+        // TODO: we need a $block here
+        // // Match by block
+        // out.push_sql(inner_prefix);
+        // out.push_identifier(BLOCK_RANGE_COLUMN)?;
+        // out.push_sql(" @> ");
+        // out.push_bind_param::<Integer, _>(&block_number)?;
```

Comment:
@lutter I'm not sure what would be the best way to pass the block number. I could drop the implementation for the `QueryFragment` and extend `walk_ast` to accept `block`.